### DISCUSSION TITLE
Renovate does not handle gitlab-ci image updates if the version comes from variable

### DISCUSSION THREAD

**Discussion by luke-sterkowicz** [renovatebot/renovate#17541] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

32.105.0

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### If you're self-hosting Renovate, tell us what version of the platform you run.

15.x

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

It's pretty standard to compose Docker image paths from variables in gitlab-ci. Renovate however doesn't honor variables and can't upgrade such images. They are recognized, listed as `undefined no version found`.
Minimal reproduction [repository](https://github.com/luke-sterkowicz/Renovate-reproduction)

### Relevant debug logs

[Dependency Dashboard](https://github.com/luke-sterkowicz/Renovate-reproduction/issues/3)
[Pull Request](https://github.com/luke-sterkowicz/Renovate-reproduction/pull/2/files)

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction repository in the bug description

**Comment by viceice** [comment]
1. this isn't a bug. renovate can't know how to resolve those variables.
2. you can try to workaround with registryAliases[^1]
3.  gitlabci manager only supports `CI_DEPENDENCY_PROXY_...` variables

[^1]: https://docs.renovatebot.com/configuration-options/#registryaliases

https://github.com/renovatebot/renovate/blob/388e9cabe21ee921536390c65a71d0c678e59514/lib/modules/manager/gitlabci/utils.ts#L19

**Reply by luke-sterkowicz** [reply]
I wasn't sure whether to treat that as a bug or new feature, though Renovate doesn't support something that Gitlab does (and I didn't find anywhere in the documentation that it wouldn't). On the other hand Renovate supports Maven properties or Dockerfile variables, so I don't see why it wouldn't Gitlab variables. Should we submit it as a feature request?

**Reply by viceice** [reply]
1. if you are unsure, open a discussion first
2. checkout registryAliases

**Reply by luke-sterkowicz** [reply]
1. I will, sorry. So having this discussion should I open a feature request or a bug?
2. In my use case, I have version tag variable set in top level gitlab-ci.yml which is propagated to an include where an image for particular jobs is set using that variable. This is similar use case to defining version properties in a topmost POM (used in child POMs). I don't really see how registryAliases would help me here.

**Reply by viceice** [reply]
why not simply move the version back to the image and let renovate handle those updates?

**Reply by luke-sterkowicz** [reply]
The include (as usual) avoids code redundancy and serves multiple other repos. The jobs with versioned image for instance serve different JDK for compiling different modules. I'd say why not use gitlab-ci features as such when they are available for you?

**Reply by luke-sterkowicz** [reply]
@viceice can we make this a new feature request?

**Reply by viceice** [reply]
🤔 It would only work for a single simple file. so `extends` or `include` for variables won't work

**Reply by luke-sterkowicz** [reply]
Ok, but why (considering it works i.e. across multiple POMs)?
